src/java_bytecode/java_object_factory.cpp

```diff
@@ -151,7 +151,7 @@ class java_object_factoryt
 /// Generates code for allocating a dynamic object. This is used in
 /// allocate_object and for allocating strings in the library preprocessing.
 /// \param target_expr: expression to which the necessary memory will be
-///   allocated
+///   allocated, its type should be pointer to `allocate_type`
```

Comment:
It's now longer a parameter of `allocate_dynamic_object_with_decl` but is still used for `allocate_dynamic_object` where it is useful if a typecast is required.